xdem/ddem.py

```diff
@@ -15,6 +17,47 @@
 from xdem._typing import MArrayf, NDArrayf
 
 
+def _mask_as_array(reference_raster: gu.Raster, mask: str | gu.Vector | gu.Raster) -> NDArrayf:
```

Comment:
I'm really wondering if this is worth having a separate function for that... It does not make much sense that it lies in ddem. But if it's only used here... Not of much help here 😉

Reply:
Yes... I guess we can leave this to be cleaned up when we revamp the `ddem` module.